aws/publisher_test.go

```diff
@@ -91,3 +91,32 @@ func TestPublish(t *testing.T) {
 		}
 	}
 }
+
+// TestDeleteTopic verifies that a sns topic can be deleted successfully with the DeleteTopic function
+func TestDeleteTopic(t *testing.T) {
+	snsMock := mockSNS{}
+	p, errPub := newPublisher(&snsMock, "foo")
+	if errPub != nil {
+		t.Errorf("expected no error from newPublisher, but got: %v", errPub)
+	}
+
+	snsMock.stubbedDeleteTopicError = errors.New("test error")
+	expected := p.DeleteTopic()
+	spiedInput := snsMock.spiedDeleteTopicInput
+	expectedTopicArn := p.topicArn
+	actualTopicArn := *spiedInput.TopicArn
+
+	// verify that DeleteTopic cathes errors
+	if expected != snsMock.stubbedDeleteTopicError {
+		t.Errorf("expected error %v, but got %v", expected, snsMock.stubbedDeleteTopicError)
```

Comment:
again, here: these are reversed. The expected is the stubbed value, the actual is what the function under test returns.